src/Pipeline.php

```diff
@@ -37,6 +37,33 @@ public function pipe(callable $pipe)
         return $this;
     }
 
+    /**
+     * Add a pipe to the pipeline before a given class.
+     *
+     * @param callable $pipe a new pipe to pass through
+     * @param string $beforeClass to class to insert the pipe before
+     *
+     * @return $this
+     */
+    public function insertBefore(callable $pipe, $beforeClass)
+    {
+        $beforePosition = null;
+        foreach ($this->pipes as $index => $callable) {
+            $class = get_class($callable);
+            if (Utils::stringEndsIn($class, $beforeClass)) {
```

Comment:
Why have we added `stringEndsIn`? Saying you need to use the fully qualified class name seems sensible - but maybe there is something I'm missing?

Reply:
I initially preferred this option, but the three implementations that I was considering were:
- Straight matching - I don't like this personally as I think it's inconvenient and too prone to false negatives, although it would be the fastest and most accurate method.
- Split the string and classname by '//', then match the last sections against each other - I like this as it allows you to go as qualified as you'd like, but it's more complex.
- String match the classname against the end of the string - It again lets you qualify as much or as little as you'd like, but is faster than splitting and matching elements of an array.  I still like this as it's flexible but fast.

Reply:
We should make this straight matching imo